google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -235,7 +235,18 @@ protected boolean projectIdRequired() {
     return true;
   }
 
-  private static AuthCredentials defaultAuthCredentials() {
+  /**
+   * Returns the default authentication credentials, or {@code null} if no default credentials could
+   * be found. This method returns the first available credentials among the following sources:
+   * <ol>
+   *   <li>App Engine credentials
+   *   <li>JSON credentials pointed to by the {@code GOOGLE_APPLICATION_CREDENTIALS} environment
```

Comment:
Is it wiser to instead give a link to the GoogleCredentials.defaultAuth method since this javadoc will become stale quickly if that method changes. I was under the impression that GoogleCredentials also supported cloud shell in addition to these other methods as well.